blocks_vertical/vertical_extensions.js

```diff
@@ -27,9 +34,29 @@ goog.require('Blockly.constants');
 goog.require('Blockly.Extensions');
 
 
+/**
+ * Helper function that generates an extension based on a category name.
+ * The generated function with set primary, secondary, and tertiary colours
+ * based on the category name.
+ * @param {String} category The name of the category to set colours for.
+ * @return {function} An extension function that sets colours based on the given
+ *     category.
+ */
 Blockly.ScratchBlocks.VerticalExtensions.colourHelper = function(category) {
+  var colours = Blockly.Colours[category];
+  if (!(colours.primary && colours.secondary && colours.tertiary)) {
```

Comment:
Should there be an error thrown if the colours cannot be found?  This code just fails silently instead.